AsyncDisplayKit/ASControlNode.mm

```diff
@@ -499,41 +506,19 @@ - (void)layout
     // determine which edges were clipped
     if (!CGSizeEqualToSize(originalRect.size, finalRect.size)) {
       
-      if (originalRect.origin.y != finalRect.origin.y) {
-        clippedEdges |= UIRectEdgeTop;
-      }
-      if (originalRect.origin.x != finalRect.origin.x) {
-        clippedEdges |= UIRectEdgeLeft;
-      }
-      if (CGRectGetMaxY(originalRect) != CGRectGetMaxY(finalRect)) {
-        clippedEdges |= UIRectEdgeBottom;
-      }
-      if (CGRectGetMaxX(originalRect) != CGRectGetMaxX(finalRect)) {
-        clippedEdges |= UIRectEdgeRight;
-      }
+      clippedEdges = [self setEdgesOfIntersectionForChildRect:originalRect parentRect:finalRect rectEdge:clippedEdges];
```

Comment:
After reading the code of both, I think it would be best to move this up to be in the same condition. You could check for the on equal size first, then run this, and then only if the parent does clip to bounds, run the check for that.

Although technically this would be less efficient and result in more method calls, the method will run very quickly and I think the code will be more readable. Because this code is a bit complex, The readability benefit is substantial.

With this change, I believe you can also eliminate the variable originalRect, and then the check for whether you should draw the edges becomes clippedEdges != UIRectEdgeNone.